packages/react/src/components/UIShell/SideNav.js

```diff
@@ -119,14 +126,30 @@ const SideNav = React.forwardRef(function SideNav(props, ref) {
     eventHandlers.onMouseLeave = () => handleToggle(false, false);
   }
 
+  useEvent(window, 'keydown', (event) => {
+    const focusedElement = document.activeElement;
+
+    if (
+      match(event, keys.Tab) &&
+      expanded &&
+      !isFixedNav &&
+      sideNavRef.current &&
+      focusedElement.id === `${prefix}--toggle_menu` &&
```

Comment:
could we achieve the same thing with a className? my concern is like you mentioned that setting the Id of the button internally prevents consumers from setting their own ids, which might be a breaking change

Reply:
Of course! I think it is a better solution also

Reply:
And I think I'll have to create a new PR to solve those merge conflicts. They are related to the typescript change in those files

Reply:
I opened this [new PR](https://github.com/carbon-design-system/carbon/pull/13809) to prevent the typescript merge conflicts @francinelucca